### PR TITLE
[22732] adds user to EVSS VSOSearch service instantiation

### PR DESCRIPTION
**Pull request by bramleyjl**
## Description of change
This PR changes the EVSS service instantiation to properly take a user as an argument so that the user's attribute values will be available as backup values in the event that `additional_headers' is missing an attribute value. It is a quick & partial reversal of [this PR](https://github.com/department-of-veterans-affairs/vets-api/pull/5793/files) that changed the EVSS service from being instantiated with a `User` instance to using authentication_headers.

For further information related to this change, see my writeup in the linked issue thread.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#22732

## Things to know about this PR

Unit tests passed, manual testing of EVSS::VSOSearch service creation with only backup user object to confirm backup functionality is working.

### CODE REVIEW

**Review by shawkey-daniel — APPROVED** [review]
LGTM

File: lib/evss/vso_search/service.rb [hunk 1]
```diff
@@ -31,16 +31,6 @@ def get_current_info(additional_headers = {})
       def request_headers(additional_headers)
         ssn = additional_headers['va_eauth_pnid'].presence || @user.ssn
         edipi = additional_headers['va_eauth_dodedipnid'].presence || @user.edipi
```
Comment (LindseySaari) [inline lib/evss/vso_search/service.rb:33]: Could there ever be a chance here still that edipi could be nil?

Comment (bramleyjl) [reply lib/evss/vso_search/service.rb]: Yes, that's definitely a possibility and we changed the logic to pull from the account instead to avoid this.

**Review by LindseySaari — APPROVED** [review]

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Pull request merged**